### PR TITLE
Pull to refresh for webViews

### PR DESCRIPTION
**Pull request by szabolcs-instructure**
Adds the option to pull to refresh `CoreWebView` and `WebView`. 

refs: MBL-16026
affects: Student, Teacher
release note: Added pull to refresh for webViews
test plan: See ticket

## Screenshots

<table>
<tr><th>K5SubjectView</th><th>Hybrid Discussion Details</th></tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

## Checklist

- [ ] Follow-up e2e test ticket created or not needed
- [ ] A11y checked
- [ ] Tested on phone
- [ ] Tested on tablet
- [ ] Approve from product or not needed

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>



<details><summary>:x: Swift lint</summary>


    ❌ /Users/vagrant/git/Core/Core/SwiftUIViews/WebView.swift:31:45: Redundant Optional Initialization Violation: Initializing an optional variable with nil is redundant. (redundant_optional_initialization)
    /Users/vagrant/git/Core/Core/SwiftUIViews/WebView.swift:76:1: error: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/Core/Core/UIViews/CoreWebView.swift:91:1: error: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    ❌ /Users/vagrant/git/Core/Core/UIViews/CoreWebView.swift:118:1: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/Core/Core/UIViews/CoreWebView.swift:139:1: error: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/Core/Core/UIViews/CoreWebView.swift:177:1: error: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    ❌ /Users/vagrant/git/Core/Core/EmbeddedWebPage/View/EmbeddedWebPageView.swift:24:1: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/Student/Student/Routes.swift:465:5: error: Opening Brace Spacing Violation: Opening braces should be preceded by a single space and on the same line as the declaration. (opening_brace)
    /Users/vagrant/git/rn/Teacher/ios/Teacher/Routes.swift:328:5: error: Opening Brace Spacing Violation: Opening braces should be preceded by a single space and on the same line as the declaration. (opening_brace)
</details>


[MBL-16026](https://instructure.atlassian.net/browse/MBL-16026)

#### Affected Apps: Student, Teacher

#### Release Note: 
Added pull to refresh for webViews
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by inst-danger** [comment]
<details><summary>Teacher Build QR Code:</summary>

</details>

File: Core/Core/SwiftUIViews/WebView.swift [hunk 1]
```diff
@@ -27,6 +27,8 @@ public struct WebView: UIViewRepresentable {
     private let source: Source?
     private var customUserAgentName: String?
     private var disableZoom: Bool = false
+    private var isPullToRefreshEnabled: Bool
+    private var pullToRefreshColor: UIColor?
```
Comment (vargaat) [inline Core/Core/SwiftUIViews/WebView.swift:31]: Does the color has any meaning if the pull to refresh is disabled? I'm thinking if we could use an enum like:
```
enum PullToRefresh {
    case disabled
    case enabled(color: UIColor)
}
```
We could change the parameter name from `isPullToRefreshEnabled` to just simply `pullToRefresh` so for example the initializer call would look like this: `WebView(url: myURL, pullToRefresh: .disabled)`

Comment (szabolcs-instructure) [reply Core/Core/SwiftUIViews/WebView.swift:31]: That's a good suggestion. Color was just an afterthought that I added before opening the PR, but you are right. An enum would make it more clear.

File: Core/Core/UIViews/CoreWebView.swift [hunk 2]
```diff
@@ -138,6 +169,17 @@ open class CoreWebView: WKWebView {
...
 
+    private func addRefreshControl(color: UIColor?) {
+        scrollView.addSubview(refreshControl)
+        scrollView.bounces = true
+        refreshControl.color = color
+    }
+
+    @objc func refreshWebView(_ sender: UIRefreshControl) {
+        reload()
+        sender.endRefreshing()
```
Comment (vargaat) [inline Core/Core/UIViews/CoreWebView.swift:180]: Would it be possible to call this in `public func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {` so the spinner would animate until the refresh navigation is finished. If `reload()` gives back a `WKNavigation` we could track exactly when the refresh navigation finishes.

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

File: Core/Core/UIViews/CoreWebView.swift (hunk 2 above)
Comment (szabolcs-instructure) [reply Core/Core/UIViews/CoreWebView.swift]: Fixed!

**Review by idevainstr — APPROVED** [review]
QA+1

**Review by davidvarga93 — APPROVED** [review]
QA + 1

**Pull request merged**